pkg/proxy/ipvs/ipset.go

```diff
@@ -102,6 +103,20 @@ func NewIPSet(handle utilipset.Interface, name string, setType utilipset.Type, i
 	hashFamily := utilipset.ProtocolFamilyIPV4
 	if isIPv6 {
 		hashFamily = utilipset.ProtocolFamilyIPV6
+		// In dual-stack both ipv4 and ipv6 ipset's can co-exist. To
+		// ensure unique names the prefix for ipv6 is changed from
+		// "KUBE-" to "KUBE-6-". The "KUBE-" prefix is kept for
+		// backward compatibility. The maximum name length of an ipset
+		// is 31 characters which must be taken into account.  The
+		// ipv4 names are not altered to minimize the risk for
+		// problems on upgrades.
+		if strings.HasPrefix(name, "KUBE-") {
+			name = strings.Replace(name, "KUBE-", "KUBE-6-", 1)
+			if len(name) > 31 {
+				klog.Warningf("Ipset name truncated; [%s] -> [%s]", name, name[:31])
```

Comment:
nit: ipset, not Ipset

Reply:
I did not know about `ipset:list`, certainly something to look into!

I think the `proxier` does not have to changed and the usage of a "ipset:list->ipset's" structure can be hidden in the ipset objects.

The only problem I see is upgrade of existing ipv4-only clusters (which is all). The existing ipset's must be "converted" to a `ipset:list` with just one (ipv4) set. This function can be feature-gated but eventually dual-stack will be GA.

When k8s is upgraded will `kube-proxy` always do a clean-up and re-install of all configuration?

If so I guess there is no problem.